linera-client/src/client_context.rs

```diff
@@ -411,6 +412,46 @@ where
         }
     }
 
+    pub async fn process_inbox_without_updating_wallet(
+        chain_client: &ChainClient<NodeProvider, S>,
+        notification_stream: &mut NotificationStream,
+    ) -> Result<Vec<ConfirmedBlockCertificate>, Error> {
+        let mut certificates = Vec::new();
+        // Try processing the inbox optimistically without waiting for validator notifications.
+        let (new_certificates, maybe_timeout) = {
+            chain_client.synchronize_from_validators().await?;
+            let result = chain_client.process_inbox_without_prepare().await;
+            result?
+        };
+        certificates.extend(new_certificates);
+        if maybe_timeout.is_none() {
+            return Ok(certificates);
+        }
+
+        loop {
+            let (new_certificates, maybe_timeout) = {
+                let result = chain_client.process_inbox().await;
+                result?
+            };
+            certificates.extend(new_certificates);
+            if let Some(timestamp) = maybe_timeout {
+                util::wait_for_next_round(notification_stream, timestamp).await
+            } else {
+                return Ok(certificates);
+            }
+        }
+    }
+
+    pub async fn start_listening_for_notifications(
+        &mut self,
+        chain_client: &ChainClient<NodeProvider, S>,
+    ) -> Result<NotificationStream, ChainClientError> {
+        // Start listening for notifications, so we learn about new rounds and blocks.
+        let (listener, _listen_handle, notification_stream) = chain_client.listen().await?;
+        self.chain_listeners.spawn_task(listener);
+        Ok(notification_stream)
+    }
```

Comment:
Do these need to be `pub`? And shouldn't they be in the `#[cfg(feature = "benchmark")]` impl below?